File: frontend/src/api/doctor.ts
// doctor 前端 API（T603）：环境诊断 15 项 + 一键修复的薄封装。
// 硬红线 4：诊断为纯读，结果以后端为准；仅 Fix 产生写（状态校准/清临时目录），且幂等可重复。
// 无宿主（纯 Vite demo）时 runDoctor 返回 null，让视图保留本地 mock 不覆盖。
import * as app from '../../bindings/phpo/app.js'
import { hasBackend } from '@/api/site'
import type { DoctorReport, DoctorStatus } from '@/types'

// runDoctor 拉取权威诊断报告；无宿主返回 null
export async function runDoctor(): Promise<DoctorReport | null> {
  if (!hasBackend()) return null
  const r = await app.DoctorRun()
  const checks = (r.checks ?? []).map((c) => ({
    id: c.id,
    title: c.title,
    status: c.status as DoctorStatus,
    detail: c.detail,
    hint: c.hint,
    fix: c.fix,
  }))
  return { checks, ok: r.ok, warnings: r.warnings, errors: r.errors }
}

// fixDoctor 执行一项一键修复（calibrate / clear_temp）；无宿主直接返回
export async function fixDoctor(id: string): Promise<void> {
  if (!hasBackend()) return
  await app.DoctorFix(id)
}

File: internal/model/doctor.go
// doctor 环境诊断视图模型（§5.7）：15 项检查逐条结果 + 汇总计数，OverviewView 消费
package model

// DoctorStatus 单项诊断结论（对应 §5.7 通过/警告/失败三态）
type DoctorStatus string

const (
	DoctorOK   DoctorStatus = "ok"   // 通过
	DoctorWarn DoctorStatus = "warn" // 警告（可继续，最小限制原则）
	DoctorErr  DoctorStatus = "err"  // 失败（可能阻断，附建议）
)

// DoctorCheck 单项诊断：ID 稳定供前端定位，Fix 非空表示支持一键修复
type DoctorCheck struct {
	ID     string       `json:"id"`
	Title  string       `json:"title"`
	Status DoctorStatus `json:"status"`
	Detail string       `json:"detail"`
	Hint   string       `json:"hint"`
	Fix    string       `json:"fix"` // "" 无 / "calibrate" 状态校准 / "clear_temp" 清空临时目录残留
}

// DoctorReport 一次诊断全量结果 + 三态计数
type DoctorReport struct {
	Checks   []DoctorCheck `json:"checks"`
	OK       int           `json:"ok"`
	Warnings int           `json:"warnings"`
	Errors   int           `json:"errors"`
}
